pkg/server/event_handler.go

```diff
@@ -234,33 +241,88 @@ func (h *EventHandler[T, P]) HTTPHandler() http.Handler {
 // [GCS notification]: https://cloud.google.com/storage/docs/pubsub-notifications#format
 func (h *EventHandler[T, P]) Handle(ctx context.Context, m pubsub.Message) error {
 	logger := logging.FromContext(ctx)
+
+	eventBytes, processErr := h.generatePmapEventBytes(ctx, m)
+
+	attr := map[string]string{}
+
+	// Use successMessenger to send to downstream if there are no
+	// errors for generating a PmapEvent.
+	// if processErr == nil {
+	// 	if err := h.successMessenger.Send(ctx, eventBytes, attr); err != nil {
+	// 		return fmt.Errorf("failed to send succuss event downstream: %w", err)
+	// 	}
+	// 	return nil
+	// }
+
+	// fmt.Println("processErr: " + processErr.Error())
+
+	// If error is a retryable error, we return err
+	// so we can start retry.
+	if processErr != nil {
+		if !errors.Is(processErr, pmaperrors.ErrNonRetryable) {
```

Comment:
I doubt in your current implementation like `processErr = fmt.Errorf("processor error: %w: failed to process object: %w", pmaperrors.ErrNonRetryable, err)`, whether the `errors.Is` will work here